test/export/InstanceExporter.test.ts

```diff
@@ -5321,6 +5321,109 @@ describe('InstanceExporter', () => {
+      );
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+    });
+
+    it('should output an error when a choice element within another element has values assigned to more than one choice type', () => {
+      // * extension[0].url = "https://example.org/SomeExt"
+      // * extension[0].valueString = "extension value is false"
+      // * extension[0].valueBoolean = false
+      const extensionUrl = new AssignmentRule('extension[0].url');
+      extensionUrl.value = 'https://example.org/SomeExt';
+      const extensionString = new AssignmentRule('extension[0].valueString');
+      extensionString.value = 'extension value is false';
+      const extensionBoolean = new AssignmentRule('extension[0].valueBoolean');
+      extensionBoolean.value = false;
+      extensionBoolean.rawValue = 'false';
+      patientInstance.rules.push(extensionUrl, extensionString, extensionBoolean);
+
+      const exported = exportInstance(patientInstance);
+      expect(exported.extension[0]).toEqual({
+        url: 'https://example.org/SomeExt',
+        valueString: 'extension value is false',
+        valueBoolean: false
+      });
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+      expect(loggerSpy.getLastMessage('error')).toMatch(
+        /Bar contains multiple choice value assignments for choice element Patient\.extension\.value\[x\]\..*File: PatientInstance\.fsh.*Line: 10 - 20\D*/s
+      );
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+    });
+
+    it('should output an error when a choice element that is a descendant of a primitive has values assigned to more than one type', () => {
+      // * gender.extension[0].url = "https://example.org/SomeExt"
+      // * gender.extension[0].valueString = "patient's gender is unknowable"
+      // * gender.extension[0].valueCode = #unknowable
+      const extensionUrl = new AssignmentRule('gender.extension[0].url');
+      extensionUrl.value = 'https://example.org/SomeExt';
+      const extensionString = new AssignmentRule('gender.extension[0].valueString');
+      extensionString.value = "patient's gender is unknowable";
+      const extensionCode = new AssignmentRule('gender.extension[0].valueCode');
+      extensionCode.value = new FshCode('unknowable');
+      patientInstance.rules.push(extensionUrl, extensionString, extensionCode);
+
+      const exported = exportInstance(patientInstance);
+      expect(exported._gender.extension[0]).toEqual({
+        url: 'https://example.org/SomeExt',
+        valueString: "patient's gender is unknowable",
+        valueCode: 'unknowable'
+      });
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+      expect(loggerSpy.getLastMessage('error')).toMatch(
+        /Bar contains multiple choice value assignments for choice element Patient\.gender\.extension\.value\[x\]\..*File: PatientInstance\.fsh.*Line: 10 - 20\D*/s
+      );
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+    });
+
```

Comment:
I think we need some negative tests too (i.e., cases where the error should not be logged). For example, we want to ensure that repeated elements that have child elements w/ choices allow for different choice types in each repetition. For example, you could write a test with multiple `Observation.component`s, each with a different `Observation.component.value[x]` type.